package/gluon-offline-ssid/luasrc/usr/bin/gluon-offline-ssid

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/lua
+
+local uci = require("simple-uci").cursor()
+local util = require 'gluon.util'
+
+function safety_exit(t)
+  io.write(t .. ", exiting with error code 2")
+  os.exit(2)
+end
+
+-- return_value is optional to return instead of the output
+function shell_exec(command, return_value)
+  if return_value then
+    return os.execute(command .. ' &> /dev/null')
+  end 
+  local file = assert(io.popen(command, 'r'))
+  local output = file:read('*all')
+  file:close()
+  return output
+end
+
+if shell_exec('pgrep -f autoupdater', true) == 0 then
+  safety_exit("autoupdater running")
+end
+
+function file_exists(name)
+   local f = io.open(name, "r")
+   return f ~= nil and io.close(f)
+end
+
+local ut = util.get_uptime()
+if ut < 60 then
+  safety_exit('less than one minute')
+end
+
+-- debug:
+function dump(o)
+   if type(o) == 'table' then
+      local s = '{ '
+      for k, v in pairs(o) do
+         if type(k) ~= 'number' then k = '"'..k..'"' end
+         s = s .. '['..k..'] = ' .. dump(v) .. ','
+      end
+      return s .. '} ' .. "\n"
+   else
+      return tostring(o)
+   end
+end
+
+-- debug:
+function vd(o)
+  print (dump(o))
+end
+
+function str_trim(s)
+   return s:match "^%s*(.-)%s*$"
+end
+
+-- only once every timeframe minutes the ssid will change to the offline-ssid
+-- (set to 1 minute if you want to change immediately every time the router gets offline)
+local minutes = tonumber(uci:get('gluon-offline-ssid', 'settings', 'switch_timeframe') or '30')
+
+-- the first few minutes directly after reboot within which an offline-ssid always may be activated
+-- (must be <= switch_timeframe)
+local first = tonumber(uci:get('gluon-offline-ssid', 'settings', 'first') or '5')
+
+-- the offline-ssid will start with this prefix use something short to leave space for the nodename
+-- (no '~' allowed!)
+local prefix = uci:get('gluon-offline-ssid', 'settings', 'prefix') or 'FF_Offline_'
```

Comment:
But what if someone fiddles with the UCI config so a value is not defined anymore? this will make sure, that a default value is still be used

Reply:
Yes, that's what I meant. Remove them from the sysupgrade file then. Or does this break something?

Reply:
it is different: in the sysupgrade file it will create a reasonable default, that will be stored in the config files for UCI.
In this file it will create a temporary reasonable value if the config files are messed up or missing.